src/writer.rs

```diff
@@ -261,6 +265,71 @@ impl<W: Write> Writer<W> {
             start_tag: BytesStart::new(name.as_ref()),
         }
     }
+
+    /// Write an arbitrary serializable type
+    ///
+    /// ```rust
+    /// # use pretty_assertions::assert_eq;
+    /// # use serde::Serialize;
+    /// # use quick_xml::events::{BytesStart, Event};
+    /// # use quick_xml::writer::Writer;
+    /// # use quick_xml::DeError;
+    /// # fn main() -> Result<(), DeError> {
+    ///
+    /// #[derive(Debug, PartialEq, Serialize)]
+    /// struct MyData {
+    ///     question: String,
+    ///     answer: u32,
+    /// }
+    ///
+    /// let data = MyData {
+    ///     question: "The Ultimate Question of Life, the Universe, and Everything".into(),
+    ///     answer: 42,
+    /// };
+    ///
+    /// let mut buffer = Vec::new();
+    /// let mut writer = Writer::new_with_indent(&mut buffer, b' ', 4);
+    ///
+    /// let start = BytesStart::new("root");
+    /// let end = start.to_end();
+    ///
+    /// writer.write_event(Event::Start(start.clone()))?;
+    /// writer.write_serializable("my_data", &data)?;
+    /// writer.write_event(Event::End(end))?;
+    ///
+    /// assert_eq!(
+    ///     std::str::from_utf8(&buffer)?,
+    ///     r#"<root>
+    ///     <my_data>
+    ///         <question>The Ultimate Question of Life, the Universe, and Everything</question>
+    ///         <answer>42</answer>
+    ///     </my_data>
+    /// </root>"#
+    /// );
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[cfg(feature = "serialize")]
+    pub fn write_serializable<T: Serialize>(
+        &mut self,
+        tag_name: &str,
+        content: &T,
+    ) -> std::result::Result<(), DeError> {
+        use crate::se::Serializer;
+
+        self.write_indent()?;
+        let indent = self.indent.clone();
+        let mut fmt = ToFmtWrite(self.get_mut());
+        let mut serializer = Serializer::with_root(&mut fmt, Some(tag_name))?;
+
+        if let Some(indent) = indent {
+            serializer.set_indent(indent);
+        }
```

Comment:
I don't think it's impossible, but I tried it and it seemed like it was becoming painful.  I can try again though.

If we don't end up borrowing here, there is little point to having `Indent::Borrow()` at all, because it has no users.

Reply:
Oh, I remember the issue.  Indeed the borrow checker was not happy about simultaneously `fmt` having a mutable reference to Writer, and `serializer` getting a reference to `Writer.indent`

Reply:
Use `&mut self.writer` directly instead of `self.get_mut()`

Reply:
Ah.  I wish the borrow checker were smart enough to see through that in the first place...